File: sistema_mercado_java/src/modelo/Cliente.java

package modelo;

import java.time.format.DateTimeFormatter;
import java.util.ArrayList;
import java.util.List;

public class Cliente extends Pessoa {

    private double cashback;
    private List<ItemVenda> carrinho = new ArrayList<>();

    public Cliente(String n, String cpf) {
        super(n, cpf);
    }

    public String mostrarCarrinho() {
        String retorno = "Carrinho: \n";
        for (ItemVenda cadaProduto : carrinho) {
            retorno += cadaProduto;
        }
        return retorno;
    }

    public void adicionarProduto(ItemVenda produto) {
        carrinho.add(produto);
    }

    public double getCashback() {
        return cashback;
    }

    public void setCashback(double cashback) {
        this.cashback += cashback;
    }

    public List<ItemVenda> getCarrinho() {
        return carrinho;
    }

    public void setCarrinho(List<ItemVenda> carrinho) {
        this.carrinho = carrinho;
    }

    @Override
    public String exibirDados() {
        String aux = "Dados do cliente";
        aux += "\nNome: " + getNome();
        aux += "\nCPF: " + getCpf();
        aux += "\nCashback acumulado: R$" + cashback;
        aux += "\nCarrinho:";
        if (carrinho != null) {
            for (ItemVenda item : carrinho) {
                aux += "\n  - " + item.getProduto().getNome();
                aux +=  ", " + item.getQuantidade() + " un.";
            }
        }
        return aux;
    }
}

File: sistema_mercado_python/modelos/Cliente.py
from modelos.Pessoa import Pessoa

class Cliente(Pessoa):
    def __init__(self, nome, cpf):
        super().__init__(nome, cpf)
        self._carrinho = []
        self._cashback = 0.0

    def adicionar_item(self, item):
        self._carrinho.append(item)

    def adicionar_cashback(self, valor):
        self._cashback += valor

    def exibir_dados(self):
        print(f"Cliente: {self._nome}, CPF: {self._cpf}, Cashback: R${self._cashback:.2f}")

    def get_carrinho(self):
        return self._carrinho

    def set_carrinho(self, carrinho):
        self._carrinho = carrinho

    def get_cashback(self):
        return self._cashback

    def set_cashback(self, cashback):
        self._cashback = cashback